### DISCUSSION TITLE
Make worstfish analyze the worst continuation instead of the best

### DISCUSSION THREAD

**Discussion by Art3141** [lichess-bot-devs/lichess-bot#909] [Ideas]
In some positions such as this:
![imagem_2024-02-09_210749855](https://github.com/lichess-bot-devs/lichess-bot/assets/143335587/b3578527-8b5f-4362-8598-90ed2965460a)
there is a selfmate. The bot plays the worst move according to stockfish (Bf7) because there is a mate in two for black. But if black does Qc6+, for example, if both sides make the worst moves, it's just a draw.
But if Be6+ was played, then there is a selfmate, Qd7 is forced, Bc7, Qxe6 is forced, Qa6+ and Qxa6# is forced.

**Comment by MarkZH** [comment]
Wouldn't a minimax algorithm using the negative of the usual board scoring value work?

Also, are you asking for changes to lichess-bot to accomplish this or just asking for advice?

**Comment by AttackingOrDefending** [comment]
We aren't the creators of WorstFish. The correct repo is [this](https://github.com/VisibleReality/lichess-worstfish).

The problem that you mentioned happens because WorstFish just searches each move for 0.1 seconds and plays the worst move if both players play the continuation perfectly. To avoid this problem, one would have to change the Stockfish itself to implement what @MarkZH suggested.

**Comment by Art3141** [comment]
Oh I'm sorry I'm new to github

**Reply by MarkZH** [reply]
No problem.
